backend/src/main/java/com/hackner/musicompass/service/ArtistReleaseService.java

```diff
@@ -0,0 +1,58 @@
+package com.hackner.musicompass.service;
+
+import com.hackner.musicompass.discogsapi.model.DiscogsMasterRelease;
+import com.hackner.musicompass.model.ArtistRelease;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.Math.pow;
+
+@Service
+public class ArtistReleaseService {
+
+    public List<ArtistRelease> getSortedReleaseList(String format, List<DiscogsMasterRelease> discogsMasterReleaseList){
+
+        List<String> discogsFormat = convertToDiscogsFormat(format);
+
+        List<ArtistRelease> formatList = discogsMasterReleaseList.stream()
+                .filter(discogsMasterRelease -> discogsMasterRelease.getFormat().stream()
+                        .anyMatch(discogsFormat::contains)) //this will get a list element from stream and call discogsFormat contains(list element)
+                .map(discogsMasterRelease -> new ArtistRelease().builder()
+                        .fullTitle(discogsMasterRelease.getFullAlbumTitle())
+                        .discogsMasterReleaseId(discogsMasterRelease.getMasterId())
+                        .format(format)
+                        .releaseYear(discogsMasterRelease.getYear())
+                        .discogsWant(discogsMasterRelease.getReleaseStats().getNumberOfWants())
+                        .discogsHave(discogsMasterRelease.getReleaseStats().getNumberOfHaves())
+                        .globalRating(calculateGlobalRating(discogsMasterRelease.getReleaseStats().getNumberOfHaves(), discogsMasterRelease.getReleaseStats().getNumberOfWants())).build())
+                .collect(Collectors.toList());
```

Comment:
This is quite hard to read, maybe extract some Methods here for clarity?!
```java
List<ArtistRelease> formatReleaseList(List<String> discogsFormat, 
                                     List<DiscogsMasterRelease> discogsMasterReleaseList) {
                return discogsMasterReleaseList.stream()
                    .filter(discogsMasterRelease -> discogsMasterRelease.getFormat().stream()
                        .anyMatch(discogsFormat::contains)) 
                   .map( discogsMasterRelease -> createArtistRelease(discogsMasterRelease))
                .collect(Collectors.toList());
                }

ArtistRelease createArtistRelease(DiscogsMasterRelease discogMasterRelease) {
return ArtistRelease.builder()
                        .fullTitle(discogsMasterRelease.getFullAlbumTitle())
                        .discogsMasterReleaseId(discogsMasterRelease.getMasterId())
                        .format(format)
                        .releaseYear(discogsMasterRelease.getYear())
                        .discogsWant(discogsMasterRelease.getReleaseStats().getNumberOfWants())
                        .discogsHave(discogsMasterRelease.getReleaseStats().getNumberOfHaves())
                        .globalRating(calculateGlobalRating(discogsMasterRelease.getReleaseStats().getNumberOfHaves(), discogsMasterRelease.getReleaseStats().getNumberOfWants())).build()
}
```

Reply:
Also try to format it better than I did in this example...

Reply:
methods are more structured now ...